expression/builtin_arithmetic.go

```diff
@@ -639,7 +639,10 @@ func (s *builtinArithmeticIntDivideDecimalSig) evalInt(row []types.Datum) (int64
 	}
 
 	ret, err := c.ToInt()
-	return ret, err == types.ErrDivByZero, errors.Trace(err)
+	if err == types.ErrOverflow {
+		return 0, false, errors.Trace(err)
+	}
+	return ret, false, nil
```

Comment:
add a comment here to explain that 
the error must be ErrTruncated which can be ignored.